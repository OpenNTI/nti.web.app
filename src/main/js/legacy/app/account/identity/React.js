const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
