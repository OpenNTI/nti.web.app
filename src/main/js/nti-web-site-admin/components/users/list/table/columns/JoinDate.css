.join-date-column {
	color: var(--secondary-grey);
	font-size: 14px;
	width: 7rem;
}

.cell {
	composes: veridically-center from './column.css';
}
