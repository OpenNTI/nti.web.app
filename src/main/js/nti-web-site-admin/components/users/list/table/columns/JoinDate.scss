@import '~@nti/style-common';

.users-table-container {
	.joindate-col {
		color: $secondary-grey;
		font-size: 14px;
		line-height: 60px;
		width: 7rem;
	}
}
