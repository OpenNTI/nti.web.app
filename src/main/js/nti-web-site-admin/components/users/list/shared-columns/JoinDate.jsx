import React from 'react';

import { scoped } from '@nti/lib-locale';
import { DateTime, Variant } from '@nti/web-core';

const t = scoped('nti-web-site-admin.users.list.shared-columns.JoinDate', {
	title: 'Join Date',
});

JoinDateColumn.Name = t('title');
JoinDateColumn.CssClassName = css`
	width: 7rem;
`;

JoinDateColumn.SortOn = 'createdTime';

JoinDateColumn.Create = props => Variant(JoinDateColumn, props);

export function JoinDateColumn({ item, getUser = x => x }) {
	const user = getUser(item);

	return (
		<DateTime
			type="body"
			date={user.getCreatedTime()}
			format={DateTime.MONTH_ABBR_DAY_YEAR}
		/>
	);
}
