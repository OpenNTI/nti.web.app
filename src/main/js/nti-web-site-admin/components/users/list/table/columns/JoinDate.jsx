import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {DateTime} from '@nti/web-commons';
import cx from 'classnames';

const t = scoped('nti-web-site-admin.components.users.list.table.columns.JoinDate', {
	headerTitle: 'Type'
});

export default class JoinDate extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
		store: PropTypes.object.isRequired,
		nonViewable: PropTypes.bool
	}

	static cssClassName = 'joindate-col';

	static Name = 'Join Date'

	// static SortKey = 'CreatedTime';

	render () {
		const {item, store} = this.props;
		return (
			<div className={cx('cell', {'row-selected': store.isSelected(item)})}>
				<DateTime className="value" date={item.getCreatedTime()} format="ll" />
			</div>
		);
	}
}
