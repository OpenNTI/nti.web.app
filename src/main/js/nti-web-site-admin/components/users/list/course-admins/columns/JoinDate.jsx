import { JoinDateColumn } from '../../shared-columns/JoinDate';

export const JoinDate = JoinDateColumn.Create({ getUser: i => i.user });
