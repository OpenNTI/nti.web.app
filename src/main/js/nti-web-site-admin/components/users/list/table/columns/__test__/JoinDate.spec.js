/* eslint-env jest */
import React from 'react';
import { create, act } from 'react-test-renderer';

import JoinDate from '../JoinDate';

describe('Site admin user table join date column test', () => {
	const verifyColumn = date => {
		let cmp;
		act(() => {
			cmp = create(
				<JoinDate
					item={{
						getCreatedTime: () => {
							return date;
						},
					}}
				/>
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	};

	test('Test date is many days ago', () => {
		// label should show date formatted as "Oct 31, 2017"
		verifyColumn(new Date('10/31/2017'));
	});
});
