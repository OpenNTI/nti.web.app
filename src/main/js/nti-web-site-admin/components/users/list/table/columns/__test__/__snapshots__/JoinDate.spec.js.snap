// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user table join date column test Test date is many days ago 1`] = `
<div>
  <time
    className="ntiText"
    dateTime="2017-10-31T00:00:00+00:00"
  >
    Oct 31, 2017
  </time>
</div>
`;
